### ISSUE TITLE
new earn section regression

### ISSUE THREAD

**Issue by nicolasburtey** [GaloyMoney/galoy-mobile#2718]
the tap to select an answer doesn't work. 

https://github.com/GaloyMoney/galoy-mobile/assets/11156370/9cee9c0f-162c-4841-9aa1-e16a27492689

**Comment by pretyflaco** [comment]
I get different types of regression:

All new chapters and lessons show up, but:

1) Only Chapter "The Origins of Money" Lesson "What is the primary and ultimate evolutionary function of collectibles" is playable  
2) However the reward offer says "Earn 0 sat" instead of "Earn 2 sats"
3) When solved correctly, an error appears "Invalid quiz question id was passed" and no reward is paid out + next lesson is not unlocked

![image](https://github.com/GaloyMoney/galoy-mobile/assets/32958103/86f8205d-6902-4d38-8a5c-0b64315f929a)
![image](https://github.com/GaloyMoney/galoy-mobile/assets/32958103/111d2431-7f2a-403f-a5e1-36257f5e59d9)

**Comment by pretyflaco** [comment]
Headline for Lesson and Question for Quiz are mixed up. Question currently shows as Headline.

201.01 
Headline: The origins of money
Quiz Question: What is the primary and ultimate evolutionary function of collectibles?

See: https://earn.blink.sv/chapter-201-the-origins-of-money/201.01-lesson-origins-of-money

**Comment by leesalminen** [comment]
> I get different types of regression:
> 
> All new chapters and lessons show up, but:
> 
> 1. Only Chapter "The Origins of Money" Lesson "What is the primary and ultimate evolutionary function of collectibles" is playable
> 2. However the reward offer says "Earn 0 sat" instead of "Earn 2 sats"
> 3. When solved correctly, an error appears "Invalid quiz question id was passed" and no reward is paid out + next lesson is not unlocked
> 
> ![image](https://user-images.githubusercontent.com/32958103/276593846-86f8205d-6902-4d38-8a5c-0b64315f929a.png) ![image](https://user-images.githubusercontent.com/32958103/276593903-111d2431-7f2a-403f-a5e1-36257f5e59d9.png)

I think all these are due to the back end not having the PR merged that adds the new sections.

**Comment by nicolasburtey** [comment]
> I think all these are due to the back end not having the PR merged that adds the new sections.

backend has been updated. at least my issue remains. it feels the click is not going through. skimming through the code it's probably coming from this diff [app/screens/earns-screen/earns-quiz.tsx ](https://github.com/GaloyMoney/galoy-mobile/commit/fc67c11c3a548da08b7d52b967f671b3e39c6966#diff-66b7257a2ac060b6e0505aa9b48b4e31d5ab1eed3552d4b99cce203ee262b9eb) ?

what was this change addressing @leesalminen ?

I'll let @pretyflaco revert back on his issue

**Comment by pretyflaco** [comment]
I cannot reproduce the error any more because the new lesson "The Origins of Money" became locked. It now shows preceding chapter "Bitcoin: Why is it special" as not completed, despite all lessons being completed. 

![image](https://github.com/GaloyMoney/galoy-mobile/assets/32958103/49ee254a-b03d-4844-8d8b-888d57ffdf75)

**Comment by UncleSamtoshi** [comment]
@pretyflaco are you still facing the issue?

**Comment by pretyflaco** [comment]
I'm on main. sth changed. Now can only access Chapter 1 & 2, while Chapter 2 shows incomplete and not possible to complete.
![image](https://github.com/GaloyMoney/galoy-mobile/assets/32958103/d116e95f-6bce-4d5d-80a6-8b203b26d67b)

![image](https://github.com/GaloyMoney/galoy-mobile/assets/32958103/48dad180-8a52-4613-b1db-0daccb70ad7f)
![image](https://github.com/GaloyMoney/galoy-mobile/assets/32958103/2c2fbd2d-5e59-4bc0-91fd-f2fa4c50289e)

**Comment by pretyflaco** [comment]
Yes, still locked out of next chapter. I'm on main.

click on "Bitcoin: Why is it special" opens the chapter and shows the lessons (all completed)
![image](https://github.com/GaloyMoney/galoy-mobile/assets/32958103/475e2d63-2786-49bc-8b19-12e84e40c02f)
![image](https://github.com/GaloyMoney/galoy-mobile/assets/32958103/7a4f318d-e66e-4bc1-9213-97befee7f526)

**Comment by leesalminen** [comment]
Wondering if this user account's state got borked when testing before the backend PR was merged, or before the funder role account was properly configured in stage.

I just created a brand new stage user (phone +50687313745) and went through the Earn section from beginning to end on the latest git commit. Works for me currently.

I don't have access to the back end of stage so I can't see that user's state.
